source/uuid.rst

```diff
@@ -0,0 +1,329 @@
+.. list-table::
+   :header-rows: 1
+   :widths: 1 1 3 1 1
+
+   * - Value
+     - Default?
+     - Encode to
+     - Decode subtype 4 to
+     - Decode subtype 3 to
+     
+   * - unspecified
+     - yes
+     - raise error 
+     - BsonBinary
+     - BsonBinary
+
+   * - standard
+     - no
+     - BSON binary subtype 4 
+     - native UUID
+     - BsonBinary
+
+   * - csharpLegacy
+     - no
+     - BSON binary subtype 3 with C# legacy byte order 
+     - BsonBinary
+     - native UUID
+
+   * - javaLegacy
+     - no
+     - BSON binary subtype 3 with Java legacy byte order 
+     - BsonBinary
+     - native UUID
+
+   * - pythonLegacy
+     - no
+     - BSON binary subtype 3 with standard byte order 
+     - BsonBinary
+     - native UUID
+
+For scenarios where the application makes the choice (e.g. a POJO with a field of type UUID), or when serializers are strongly typed and are constrained to always return values of a certain type, the driver will raise an exception in cases where otherwise it would be required to decode to a different type (e.g. BsonBinary instead of UUID or vice versa).
+
+Note also that none of the above applies when decoding to strictly typed maps, e.g. a Map<String, BsonValue> like Java or .NET's BsonDocument class.  In those cases the driver is always decoding to BsonBinary, and applications would use the asUuid methods to explicitly convert from BsonBinary to UUID.
+
+
+Implementation Notes
+--------------------
+
+Since changing the default UUID representation can reasonably be considered a backwards-breaking change, drivers that implement the full specification should stage implementation according to semantic versioning guidelines.  Specifically, support for this specification can be added to a minor release, but with several exceptions: 
+
+The default UUID representation should be left as is (e.g. JAVA_LEGACY for the Java driver) rather than be changed to UNSPECIFIED.  In a subsequent major release, the default UUID representation can be changed to UNSPECIFIED (along with appropriate documentation indicating the backwards-breaking change).
+Subtype 4 BSON Binary values should continue to be decoded to native UUID values regardless of UUID representation
+
+
+Test Plan
+=========
+
+The test plan consists of a series of prose tests.  They all operate on the same UUID, with the String representation of "00112233-4455-6677-8899-aabbccddeeff".
+
+Explicit encoding
+-----------------
+
+1. Create a BsonBinary instance with the given UUID
+   
+   a. Assert that the BsonBinary instance's subtype is equal to 4 and data equal to the hex-encoded string "00112233445566778899AABBCCDDEEFF"
+
+2. Create a BsonBinary instance with the given UUID and UuidRepresentation equal to STANDARD
+
+   a. Assert that the BsonBinary instance's subtype is equal to 4 and data equal to the hex-encoded string "00112233445566778899AABBCCDDEEFF"
+
+3. Create a BsonBinary instance with the given UUID and UuidRepresentation equal to JAVA_LEGACY
```

Comment:
If it's not too much trouble, can the java and csharp examples be swapped to maintain lexicographical order?

Reply:
STANDARD is still out of order though.  Do you want that moved to?

In general, though, how is lexicographical ordering helpful? I'd rather not change this now and have to go back and updated anyone's existing ordering in test files, if they referenced these numbers.

Reply:
The way it is is fine then.